docx/header.py

```diff
@@ -16,4 +18,28 @@ class Header(ElementProxy):
     The default page header of a section.
     """
 
-    __slots__ = ()
+    __slots__ = ('type')
+
+    def __init__(self, element, parent=None, type=WD_HEADER_TYPE.PRIMARY):
+        super(Header, self).__init__(element, parent=parent)
+        self.type = type
```

Comment:
* All private variables have a leading underscore (`_type` rather than `type`)
* parent and type are both required, so no need to provide defaults.
* added `self._sectPr = element` for clarity, `self._element` is retained as it is provided by ElementProxy and occasionally is useful for general-purpose methods that don't have any way of knowing the specific element spelling.
* created and inherited from `_BaseHeaderFooter` since is_linked_to_previous will be general to all header/footer types.